### PR TITLE
Sql: Add support for VirtualNetworkRules

### PR DESCRIPTION
**Pull request by jaredmoo**
## Description
Added VirtualNetworkRules feature including scenario tests.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by jaredmoo** [comment]
Jenkins infra error, rerunning: http://azuresdkci.cloudapp.net/job/NetCore-SdkCI/480/

File: src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            SqlManagementTestUtilities.RunTestInNewV12Server(suiteName, "TestGetAndListVirtualNetworkRule", testPrefix, (resClient, sqlClient, resourceGroup, server) =>
+            {
+                // Create Firewall Rules
+                //
+                Dictionary<string, VirtualNetworkRule> rules = new Dictionary<string, VirtualNetworkRule>();
+
+                string vnetfirewallRuleName = SqlManagementTestUtilities.GenerateName(testPrefix);
+                VirtualNetworkRule rule = new VirtualNetworkRule()
+                {
+                    VirtualNetworkSubnetId = string.Format("/subscriptions/d513e2e9-97db-40f6-8d1a-ab3b340cc81a/resourceGroups/TestVnetSdk/providers/Microsoft.Network/virtualNetworks/vnetSdkTest/subnets/subnet1")
```
Comment (markcowl) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: Isn't this just a hard-coded value that will make it more difficult to re-record the test?

File: src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            string suiteName = this.GetType().FullName;
+            SqlManagementTestUtilities.RunTestInNewV12Server(suiteName, "TestCreateAndDropVirtualNetworkRule", testPrefix, (resClient, sqlClient, resourceGroup, server) =>
+            {
+                // Create and validate Firewall Rule
+                //
+                string vnetfirewallRuleName = SqlManagementTestUtilities.GenerateName(testPrefix);
+
+                VirtualNetworkRule toCreate = new VirtualNetworkRule()
+                {
+                    VirtualNetworkSubnetId = string.Format("/subscriptions/d513e2e9-97db-40f6-8d1a-ab3b340cc81a/resourceGroups/TestVnetSdk/providers/Microsoft.Network/virtualNetworks/vnetSdkTest/subnets/subnet1")
```
Comment (markcowl) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: same question

File: src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            string suiteName = this.GetType().FullName;
+            SqlManagementTestUtilities.RunTestInNewV12Server(suiteName, "TestCreateAndUpdateVirtualNetworkRule", testPrefix, (resClient, sqlClient, resourceGroup, server) =>
+            {
+                // Create Firewall Rule and Validate
+                //
+                string vnetfirewallRuleName = SqlManagementTestUtilities.GenerateName(testPrefix);
+
+                VirtualNetworkRule toCreate = new VirtualNetworkRule()
+                {
+                    VirtualNetworkSubnetId = string.Format("/subscriptions/d513e2e9-97db-40f6-8d1a-ab3b340cc81a/resourceGroups/TestVnetSdk/providers/Microsoft.Network/virtualNetworks/vnetSdkTest/subnets/subnet1")
```
Comment (markcowl) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: same question

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs (hunk 1 above)
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: Yes this is bad long-term, the problem is that the network RP's API to create these virtual networks is not fully working in production so it's not possible to automate right now. We'll make an issue assigned to @dnayantara to update this asap.

Comment (dnayantara) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: Yes! Unfortunately Networking team's .NET SDK code isnt public yet and they are also aiming for the July release. Once that is out, I plan to change these tests and re-record them using the Networking teams .Net sdk client code.

Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/VirtualNetworkRulesCrudScenarioTests.cs]: Created issue #3388 and added TODO in code.

**Review by markcowl — APPROVED** [review]

**Pull request merged**